### PR TITLE
enhancement(stackdriver_metrics sink): Implement batching

### PR DESCRIPTION
**Pull request by YoEight**

### CODE REVIEW

**Comment by YoEight** [comment]
I'm having trouble fixing the docs, can someone help me out? This is what I got this and don't understand what I need to do:
```
components.sinks.gcp_stackdriver_metrics.configuration.max_batch_size.type: 2 errors in empty disjunction::
    ./website/cue/reference.cue:285:5
make: *** [Makefile:639: check-docs] Error 1
```

File: website/cue/reference/components/sinks/gcp_stackdriver_metrics.cue [hunk 1]
```diff
@@ -106,6 +106,13 @@ components: sinks: gcp_stackdriver_metrics: {
 				syntax:  "literal"
 			}
 		}
+		max_batch_size: {
+			common:      false
+			description: "Maximum time series batch size used when sending metrics to GCP."
+			required:    false
+			warnings: []
+			type: uint64|null
```
Comment (lucperkins) [inline website/cue/reference/components/sinks/gcp_stackdriver_metrics.cue:114]: For the `type` here, you need to provide an object. Here's an example:

```cue
type: uint: {
    default: 200
    unit: "bytes"
}
```

Comment (YoEight) [reply website/cue/reference/components/sinks/gcp_stackdriver_metrics.cue]: Thanks! It worked.

File: src/sinks/gcp/stackdriver_metrics.rs [hunk 2]
```diff
@@ -21,11 +20,7 @@ pub struct StackdriverConfig {
     pub credentials_path: Option<String>,
     #[serde(default = "default_metric_namespace_value")]
     pub default_namespace: String,
-    #[serde(default)]
-    pub request: TowerRequestConfig,
-    #[serde(default)]
-    pub batch: BatchConfig,
-    pub tls: Option<TlsOptions>,
```
Comment (bruceg) [inline src/sinks/gcp/stackdriver_metrics.rs:28]: At minimum, the `request` and `tls` options will need to be reintroduced, as they will become part of the component specification. Ideally, `max_batch_size` should use the `batch` options above as well for consistency with other components.

File: src/sinks/gcp/stackdriver_metrics.rs [hunk 3]
```diff
@@ -99,74 +66,106 @@ impl SinkConfig for StackdriverConfig {
 struct HttpEventSink {
     config: StackdriverConfig,
     started: DateTime<Utc>,
+    client: reqwest::Client,
```
Comment (bruceg) [inline src/sinks/gcp/stackdriver_metrics.rs:69]: Our standard practice has been to use `hyper` directly for HTTP clients, and we would prefer to continue that pattern.

**Review by bruceg — CHANGES_REQUESTED** [review]
Thank you for contributing to this development. We have been heading in this direction with other sinks, notably `aws_s3` in #8884 and `datadog_logs` in #8825. We have noted a few concerns with this PR that will need to be addressed before doing a more substantial review.

In addition to the comments below, we are concerned that this time-based batching will not work correctly if the source component(s) are not continuously feeding events to drive the core loop.

**Comment by jszwedko** [comment]
Closing since this is stale, but please feel free to reopen if you pick this work back up. Thanks for the start of this contribution!

**Pull request closed without merging**